File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/squirmy/GamepadTest.java
package org.firstinspires.ftc.teamcode.squirmy;

import com.qualcomm.hardware.lynx.LynxModule;
import com.qualcomm.robotcore.eventloop.opmode.OpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import com.qualcomm.robotcore.hardware.DcMotorEx;

import org.firstinspires.ftc.robotcore.external.navigation.CurrentUnit;

import java.util.List;

@TeleOp(name = "Squirmy Gamepad")
public class GamepadTest extends OpMode {

    double x = 0.0;
    double loopTime = 0.0;

    @Override
    public void init() {

    }

    @Override
    public void loop() {
        // try caching values
        x = 0.0;
        x = gamepad1.left_stick_y;
        x = gamepad1.left_stick_x;
        x = gamepad1.right_stick_x;
        x = gamepad1.right_stick_y;
        x = gamepad1.touchpad_finger_1_x;
        x = gamepad1.touchpad_finger_2_x;
        x = gamepad1.left_trigger;
        x = gamepad1.right_trigger;
        double loop = System.nanoTime();
        telemetry.addData("hz", 1000000000 / (loop - loopTime));
        telemetry.addData("loop time", (loop - loopTime)/1000000);
        loopTime = loop;
        telemetry.update();
    }
}

File: TeamCode/src/main/kotlin/org/firstinspires/ftc/teamcode/seventh/opmode/test/GamepadTest.kt
package org.firstinspires.ftc.teamcode.seventh.opmode.test

import com.arcrobotics.ftclib.gamepad.GamepadEx
import com.arcrobotics.ftclib.gamepad.GamepadKeys
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.seventh.opmode.teleop.Gigapad
import org.firstinspires.ftc.teamcode.seventh.robot.hardware.Robot

@TeleOp(name = "Gamepad Test")
class GamepadTest : LinearOpMode() {
    override fun runOpMode() {
        val gamepad = Gigapad(gamepad1)
        Robot.init(hardwareMap, telemetry, gamepad, null)

        waitForStart()
        while (opModeIsActive() && !isStopRequested) {
            Robot.read()
            telemetry.addData("dt", Robot.getDt()*1000)
            telemetry.addData("left x", gamepad.leftX)
            telemetry.addData("left y", gamepad.leftY)
            telemetry.addData("right x", gamepad.rightX)
            telemetry.addData("right y", gamepad.rightY)
            telemetry.addData("left trigger", gamepad.getTrigger(GamepadKeys.Trigger.LEFT_TRIGGER))
            telemetry.addData("right trigger", gamepad.getTrigger(GamepadKeys.Trigger.RIGHT_TRIGGER))
            telemetry.addData("dpad up", gamepad.isDown(GamepadKeys.Button.DPAD_UP))
            telemetry.addData("dpad down", gamepad.isDown(GamepadKeys.Button.DPAD_DOWN))
            telemetry.addData("dpad left", gamepad.isDown(GamepadKeys.Button.DPAD_LEFT))
            telemetry.addData("dpad right", gamepad.isDown(GamepadKeys.Button.DPAD_RIGHT))
            telemetry.update()
        }
    }
}